tests/functions.py

```diff
@@ -56,17 +77,38 @@ def recurse(directory, odds):
         # If the file is a broken symlink and with odd chance
         elif not exists(full_path) and random() < odds:
             # Timeouts the download or a hanging authentification
-            with timeout(3): 
-                msg = api.get(path=full_path, on_failure="ignore", return_type="item-or-list")
-                
+            with timeout(3):
+                msg = api.get(
+                    path=full_path, on_failure="ignore", return_type="item-or-list"
+                )
+
                 # Check for authentication
-                if isinstance(msg, dict) and msg["status"] == "error" and "unable to access" not in msg["message"].lower():
-                    return "Cannot download file and didn't hit authentication request for file: " + full_path
```

Comment:
This needs to be updated as follows:
1. If dataset doesn't require authentication (check that from DATS model) then just print "Cannot download file (dataset doesn't require authentication)".
2. If dataset requires authentication then print "Cannot download file (dataset requires authentication, make sure that variables X, Y and Z are defined in Travis)"